src/services/codefixes/inferFromUsage.ts

```diff
@@ -150,35 +155,67 @@ namespace ts.codefix {
             return;
         }
 
-        const types = inferTypeForParametersFromUsage(containingFunction, sourceFile, program, cancellationToken) ||
-            containingFunction.parameters.map(p => isIdentifier(p.name) ? inferTypeForVariableFromUsage(p.name, program, cancellationToken) : undefined);
+        const symbols = inferTypeForParametersFromUsage(containingFunction, sourceFile, program, cancellationToken) ||
+            containingFunction.parameters.map(p => ({
+                declaration: p,
+                type: isIdentifier(p.name) ? inferTypeForVariableFromUsage(p.name, program, cancellationToken) : undefined
+            } as ParameterInference));
         // We didn't actually find a set of type inference positions matching each parameter position
-        if (!types || containingFunction.parameters.length !== types.length) {
+        if (containingFunction.parameters.length !== symbols.length) {

```

Comment:
It looks like `symbols` is the result of `map` so it should always be the case? Replacing this with `Debug.assert(containingFunction.parameters.length === symbols.length);` showed no errors in any tests.

Reply:
Yep! Good catch.